fingerprint.c

```diff
@@ -222,27 +242,59 @@ void *auth_thread_loop(void *arg)
 
                     msg.data.authenticated.hat = hat;
 
+                    setState(sdev, STATE_IDLE);
```

Comment:
Maybe it's better to not set the state until just before exit from the function?

Reply:
@spiiroin there is an issue where if you let the function return for auth / enroll and the state is isnt idle the state machine will try and run the auth / enroll again. In the case of the enroll that will crash the TZ app